sdk/web-pubsub/web-pubsub-express/src/cloudEventsDispatcher.ts

```diff
@@ -100,54 +103,70 @@ export class CloudEventsDispatcher {
       console.log(receivedEvent);
     }
 
+    const stringifyStates = this.stringifyStates;
     switch (eventType) {
       case EventType.Connect: {
         const connectRequest = receivedEvent.data as ConnectRequest;
-        connectRequest.context = this.GetContext(receivedEvent, origin);
-        this.eventHandler.handleConnect!(connectRequest, {
-          success(res?: ConnectResponse): void {
-            response.statusCode = 200;
-            if (res === undefined) {
-              response.end();
-            } else {
-              response.setHeader("Content-Type", "application/json; charset=utf-8");
-              response.end(JSON.stringify(res));
-            }
-          },
-          fail(code: 400 | 401 | 500, detail?: string): void {
-            response.statusCode = code;
-            response.end(detail ?? "");
-          }
-        });
+        connectRequest.context = this.getContext(receivedEvent, origin);
+        this.eventHandler.handleConnect!(
+          connectRequest,
+          (function() {
+            let states: Record<string, any> = connectRequest.context.states;
+            let modified = false;
+            return {
+              setState(name: string, value: any): ConnectResponseHandler {
+                states[name] = value;
+                modified = true;
+                return this;
```

Comment:
if you assigned the anonymous response object to a variable before returning it, you wouldn't have to rely on `this` here